EIPS/eip-2159.md

```diff
@@ -0,0 +1,56 @@
+---
+eip: 2159
+title: Common Prometheus Metrics Names for Clients
+author: Adrian Sutton (@ajsutton)
+discussions-to: https://ethereum-magicians.org/t/common-chain-metrics/3415/2
+status: Draft
+type: Informational
```

Comment:
So is this an ERC, Interface or Informational?

Is this proposing that clients when integrate with Prometheus they should use the names below? What does integrating with Prometheus means?

In short: does this proposes something clients would implement or is this implemented by third parties when integrating a client into Prometheus?

Reply:
This is something that clients would implement, but exactly how they integrate with Prometheus is up to them - there are a few ways a client and Prometheus could be configured to work together and no real benefit in trying to standardise it.

What is useful is if a client is sending metrics to Prometheus (via whatever means), that these few properties have the same names and meanings.  Then you can create a dashboard like https://grafana.com/api/dashboards/10273/images/6473/image to monitor groups of nodes.

I'm not exactly sure what category this should go under.  Interface is a possibility and would definitely be the case if it were standardising how to integrate with Prometheus.  I've put it as Informational here because its essentially a recommended convention. It's not trying to be the definitive spec on metrics for clients and its entirely optional whether clients choose to support it or not. It's just a little more convenient if they do.

I don't mind which category it winds up in though.

Reply:
Having read the relevant section on EIP-1 again as well as looking what actual EIPs were merged under Interface (https://eips.ethereum.org/interface) I think the right category would be `Interface` for this.

Reply:
Interface it is.